cmd/example-hook-sidecar/smbios.go

```diff
@@ -63,12 +67,41 @@ func (s infoServer) Info(ctx context.Context, params *hooksInfo.InfoParams) (*ho
 				Name:     hooksInfo.OnDefineDomainHookPointName,
 				Priority: 0,
 			},
+			{
+				Name:     hooksInfo.ShutdownHookPointName,
+				Priority: 0,
+			},
 		},
 	}, nil
 }
 
 type v1alpha1Server struct{}
 type v1alpha2Server struct{}
+type v1alpha3Server struct {
+	done chan struct{}
+}
+
+func (s v1alpha3Server) Shutdown(ctx context.Context, params *hooksV1alpha3.ShutdownParams) (*hooksV1alpha3.ShutdownResult, error) {
+	log.Log.Info(onShutdownMessage)
+	s.done <- struct{}{}
+	return &hooksV1alpha3.ShutdownResult{}, nil
+}
+
+func (s v1alpha3Server) OnDefineDomain(ctx context.Context, params *hooksV1alpha3.OnDefineDomainParams) (*hooksV1alpha3.OnDefineDomainResult, error) {
+	log.Log.Info(onDefineDomainLoggingMessage)
+	newDomainXML, err := onDefineDomain(params.GetVmi(), params.GetDomainXML())
+	if err != nil {
+		return nil, err
+	}
+	return &hooksV1alpha3.OnDefineDomainResult{
+		DomainXML: newDomainXML,
+	}, nil
+}
+
+func (s v1alpha3Server) PreCloudInitIso(_ context.Context, _ *hooksV1alpha3.PreCloudInitIsoParams) (*hooksV1alpha3.PreCloudInitIsoResult, error) {
+	// Needs to be created to satisfy interface
+	panic("Not implemented. Not exported. Should not be called.")
```

Comment:
I am not completely familiar with this, so you might have a good reason for doing this, but could you explain your reasoning behind panicking here and not implementing this? Thanks :)

Reply:
This is not relevant anymore as `sidecar-shim` implements the `PreCloudInitIso`.

> could you explain your reasoning behind panicking here and not implementing this?

Yes. The smbios sidecar is only an example that changes the `baseBoardManufacturer` to `Radical Edward`. In order to do this, it needs to change libvirt's domain xml prior the VM start, using the `OnDefineDomain` hook callback.

It does not need other hooks.

We need to implement them for the sake of hook's interface. When I say we don't expose `PreCloudInitIso`, this is related to `Info` method, that provides to `virt-handler` or `hookManager` the functions/callbacks this sidecar expose.